packages/11ty/_includes/components/object-card.webc

```diff
@@ -1,48 +1,73 @@
-<script webc:type="js">
-// TODO: get `ignoredAttributes` from `objects-catalog.webc`
-const ignoredAttributes = ['name', 'thumbnail', 'title'];
-const renderFields = () => {
-  return Object
-    .entries(this.webc.attributes)
-    .reduce((fields, [key, value]) => {
+<a class="object-card" href="this.webc.attributes.url">
+  <div class="object-card__content" @html="renderFields({ ...this.webc.attributes, $data: this.$data })"></div>
+</a>
+
+<script webc:setup>
+const renderFields = (objectCardAttributes) => {
+  const { $data, figures } = objectCardAttributes;
+  const ignoredAttributes = ['uid', 'url'];
+  const isIgnoredAttribute = (key) => !ignoredAttributes.includes(key);
+  const attributeOrder = $data.object_card || $data.objects.object_card || Object.keys(objectCardAttributes).filter(isIgnoredAttribute);
+  const attributes = attributeOrder.map((key) => ({ name: key, value: objectCardAttributes[key]}));
+  const firstTextAttribute = attributes.find(({ name }) => name !== 'thumbnail');
+  const alt = () => typeof thumbnail === 'object' ? thumbnail.alt : '';
+  /**
+   * If the object has figures, set thumbnail as the first image figure thumbnail.
+   * Otherwise use a thumbnail defined in object data
+   */
+  const src = () => {
+    const thumbnail = figures
+      ? figures.find(({ mediaType }) => mediaType === 'image').thumbnail
+      : objectCardAttributes.thumbnail
+
+    const src = typeof thumbnail === 'object'
+      ? thumbnail.src
+      : thumbnail
+
+    /**
+     * If no IIIF images, point to the raw assets directory
+     */
+    return src && !src.startsWith('/iiif')
+      ? `${this.config.figures.imageDir}/${src}`
+      : src;
+  }
```

Comment:
I moved all of this thumbnail `src`-mangling logic from `objects-catalog` to here to clean that component up